src/sentry/grouping/enhancer/__init__.py

```diff
@@ -474,6 +496,126 @@ def visit_quoted_ident(self, node, children):
         return node.match.groups()[0].lstrip("!")
 
 
+def _update_frames_from_cached_values(
+    frames: Sequence[dict[str, Any]], cache_key: str, platform: str
+) -> bool:
+    """
+    This will update the frames of the stacktrace if it's been cached.
+    Returns if the merged has correctly happened.
+    """
+    frames_changed = False
+    # XXX: Try the fallback value
+    changed_frames_values = cache.get(cache_key, {})
+    # This helps tracking changes in the hit/miss ratio of the cache
+    metrics.incr(
+        "save_event.stacktrace.cache.get",
+        tags={"success": bool(changed_frames_values), "platform": platform},
+    )
+    if changed_frames_values:
+        try:
+            for frame, changed_frame_values in zip(frames, changed_frames_values):
+                frame["in_app"] = changed_frame_values["in_app"]
+                set_path(frame, "data", "category", value=changed_frame_values["category"])
+
+            logger.info("We have merged the cached stacktrace to the incoming one.")
+
+            frames_changed = True
+        except Exception:
+            logger.exception(
+                "We have failed to update the stacktrace from the cache. Not aborting execution."
+            )
+
+    metrics.incr(
+        "save_event.stacktrace.merged_cached_values",
+        tags={"success": frames_changed, "platform": platform},
+    )
+    return frames_changed
+
+
+def _cache_changed_frame_values(
+    frames: Sequence[dict[str, Any]], cache_key: str, platform: str
+) -> None:
+    """Store in the cache the values which have been modified for each frame."""
+    caching_succeeded = False
+    # Check that some other event has not already populated the cache
+    if cache.get(cache_key):
+        return
+
+    try:
+        # XXX: A follow up PR will be required to make sure that only a whitelisted set of parameters
+        # are allowed to be modified in apply_modifications_to_frame, thus, not falling out of date with this
+        changed_frames_values = [
+            {
+                "in_app": frame.get("in_app"),  # Based on FlagAction
+                "category": get_path(frame, "data", "category"),  # Based on VarAction's
+            }
+            for frame in frames
+        ]
+        cache.set(cache_key, changed_frames_values)
```

Comment:
Something to consider: we might want to lower the cache timeout but refresh it if still in us rather than or have some sort of one-hit-wonder prevention here. I'm guessing that most cache keys will be completely useless